docs/datastorage/capabilities.md

```diff
@@ -0,0 +1,176 @@
+The Capability System
+==========================
+
+Capabilities allow exposing features in a dynamic and flexible way, without having to resort to directly implementing many interfaces.
+
+In general terms, each capability provides a feature in the form of a class or interface, alongside with a default implementation which can be requested, and a storage handler for at least this default implementation. The storage handler can support other implementations, but this is up to the capability implementor, so look it up in their documentation before trying to use the default storage with non-default implementations.
+
+Forge adds capability support to TileEntities, Entities, and ItemStacks, which can be exposed either by attaching them through an event or, with the exception of ItemStacks, by overriding he capability methods in your own implementations of the entities. This will be explained in more detail in the following sections.
+
+Using an Existing Capability
+----------------------------
+
+This most notably applies to the Item Handler capability added by Forge, which replaces the vanilla IInventory and ISidedInventory with an automation-friendly system.
+
+As mentioned earlier, TileEntities, Entities, and ItemStacks implement the capability provider feature, through the `ICapabilityProvider` interface. This interface adds two methods, `hasCapability` and `getCapability`, which can be used to query the capabilities present in the objects.
+
+In order to obtain a capability, you will need to refer it by its unique instance. In the case of the Item Handler, this capability is primarily stored in `CapabilityItemHandler.ITEM_HANDLER_CAPABILITY`, but it is possible to get an instance reference by using the `@CapabilityInject` annotation. This annotation can also be used on a method in order to get notified when a capability is registered, so that certain features can be enabled conditionally. Please refer to the javadocs for this annotation for details on usage.
+
+Both the `hasCapability` and `getCapability` methods have a second parameter, of type EnumFacing, which can be used in the TileEntities to request the specific instance for that one face. If passed `null`, a general capability instance will be requested instead. The return type of `getCapability` will correspond to the type declared in the capability passed to the method. For the Item Handler capability, this is indeed `IItemHandler`.
+
```

Comment:
Something along the lines of 'null' is considered the 'self', or 'unknown' value and should be treated as if the request is coming from inside the block, or potentially across dimensions. It should return a interface that doesn't care about side, or just pick a side to treat it as.